src/rust/engine/src/externs.rs

```diff
@@ -717,25 +718,15 @@ impl BufferBuffer {
   }
 
   pub fn to_map(&self, name: &'static str) -> Result<BTreeMap<String, String>, String> {
-    self
+    let strings = self
       .to_strings()
-      .map_err(|err| format!("Error decoding UTF8 from {}: {}", name, err))?
-      .into_iter()
-      .map(|s| {
-        let mut parts: Vec<_> = s.splitn(2, '=').collect();
-        if parts.len() != 2 {
-          return Err(format!(
-            "Got invalid {} - must be of format key=value but got {}",
-            name, s
-          ));
-        }
-        let (value, key) = (
-          parts.pop().unwrap().to_owned(),
-          parts.pop().unwrap().to_owned(),
-        );
-        Ok((key, value))
-      })
-      .collect()
+      .map_err(|err| format!("Error decoding UTF8 from {}: {}", name, err))?;
+
+    if strings.len() % 2 != 0 {
+      return Err(format!("Map for {} had an odd number of elements", name));
+    }
+
+    Ok(strings.into_iter().tuple_windows::<(_, _)>().collect())
```

Comment:
noice 👍

Reply:
The `tuples` method is more correct than `tuple_windows` here.